src/fileutil.c

```diff
@@ -63,6 +63,15 @@ rs_file_open(char const *filename, char const *mode)
 	    return stdin;
     }
 
+    if (!force && is_write) {
+	if ((f = fopen(filename, "rb"))) {
```

Comment:
fdopen() is POSIX not ANSI so it hurts portability.
ftruncate() is again problematic, it fails silently on Windows if file size >2 GB.

Closing and reopening really seems like the simplest solution :-)